packages/trackerless-network/src/NetworkNode.ts

```diff
@@ -0,0 +1,165 @@
+    }
+
+    addMessageListener<T>(cb: (msg: StreamMessage<T>) => void): void {
+        this.stack.getStreamrNode().on('newMessage', (msg) => {
+            const translated = StreamMessageTranslator.toClientProtocol<T>(msg)
+            return cb(translated)
+        })
+    }
+
+    setStreamPartEntryPoints(streamPartId: StreamPartID, contactPeerDescriptors: PeerDescriptor[]): void {
+        this.stack.getStreamrNode()!.setStreamPartEntryPoints(streamPartId, contactPeerDescriptors)
+    }
+
+    removeMessageListener<T>(cb: (msg: StreamMessage<T>) => void): void {
+        if (this.stopped) {
+            return
+        }
+        this.stack.getStreamrNode().off('newMessage', (msg) => {
+            const translated = StreamMessageTranslator.toClientProtocol<T>(msg)
+            return cb(translated)
+        })
+    }
+
+    async subscribeAndWaitForJoin(
+        streamPartId: StreamPartID,
+        timeout?: number,
+        expectedNeighbors?: number
+    ): Promise<number> {
+        if (this.stack.getStreamrNode()!.isProxiedStreamPart(streamPartId, ProxyDirection.PUBLISH)) {
+            throw new Error(`Cannot subscribe to ${streamPartId} as proxy publish connections have been set`)
+        }
+        return this.stack.getStreamrNode().waitForJoinAndSubscribe(streamPartId, timeout, expectedNeighbors)
+    }
+
+    async waitForJoinAndPublish(streamMessage: StreamMessage, timeout?: number): Promise<number> {
+        const streamPartId = streamMessage.getStreamPartID()
+        const msg = StreamMessageTranslator.toProtobuf(streamMessage)
+
+        if (this.stack.getStreamrNode()!.isProxiedStreamPart(streamPartId, ProxyDirection.SUBSCRIBE)) {
+            throw new Error(`Cannot publish to ${streamPartId} as proxy subscribe connections have been set`)
+        }
+
+        return this.stack.getStreamrNode().waitForJoinAndPublish(streamPartId, msg, timeout)
+    }
+
+    unsubscribe(streamPartId: StreamPartID): void {
+        if (this.stopped) {
+            return
+        }
+        this.stack.getStreamrNode().unsubscribeFromStream(streamPartId)
+    }
+
+    getNeighborsForStreamPart(streamPartId: StreamPartID): ReadonlyArray<string> {
+        return this.hasStreamPart(streamPartId)
+            ? this.stack.getStreamrNode().getStream(streamPartId)!.layer2.getTargetNeighborStringIds()
+            : []
+    }
+
+    hasStreamPart(streamPartId: StreamPartID): boolean {
+        return this.stack.getStreamrNode().hasStream(streamPartId)
+    }
+
+    hasProxyConnection(streamPartId: StreamPartID, contactNodeId: string, direction: ProxyDirection): boolean {
+        return this.stack.getStreamrNode()!.hasProxyConnection(streamPartId, contactNodeId as PeerIDKey, direction)
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    getRtt(_nodeId: string): number | undefined {
+        throw new Error('Not implemented')
```

Comment:
Will the trackerless network support this functionality, or should we remove the method?

Reply:
Most likely not as this feature was only used in the brubeck miner plugin

Reply:
Removed in https://github.com/streamr-dev/network/pull/1897